packages/workbox-cache-expiration/CacheExpirationManager.mjs

```diff
@@ -187,6 +192,47 @@ class CacheExpirationManager {
       await this._timestampModel.deleteUrl(url);
     }
   }
+
+  /**
+   * Update the timestamp for the given URL. This ensures the when
+   * removing entries based on maximum entries, most recently used
+   * is accurate or when expiring, the timestamp is up-to-date.
+   *
+   * @param {string} url
+   */
+  async updateTimestamp(url) {
+    if (process.env.NODE_ENV !== 'production') {
+      core.assert.isType(url, 'string', {
+        moduleName: 'workbox-cache-expiration',
+        className: 'CacheExpirationManager',
+        funcName: 'updateTimestamp',
+        paramName: 'url',
+      });
+    }
+
+    const urlObject = new URL(url, location);
+    urlObject.hash = '';
+
+    await this._timestampModel.setTimestamp(urlObject.href, Date.now());
+  }
+
+  /**
+   * Can be used to check if a URL has expired or not before it's used.
+   *
+   * Note: This method will not remove the cached entry, call
+   * `expireEntries()` to remove indexedDB and Cache entries.
+   *
+   * @param {string} url
+   * @return {boolean}
+   */
+  async isURLExpired(url) {
```

Comment:
Is this currently being used? If not, how do you envision it being used?

(This effectively moves some of our discussion about `isResponseFresh()` from email to GitHub.)

Reply:
It will be a function developers of the module can use if they want.

The plugin I'm just go to add in the current behavior and leave it as it.

Reply:
👍